docs/datasets/index.rst

```diff
@@ -52,10 +52,106 @@ Note that in Gammapy, 2D image analyses are done with 3D cubes with a single
+are simply summed,
+while for energy dispersion and point spread function an average is computed after
+weighing by the exposure.
+
+For the model evaluation, an important factor that needs to be accounted for is
+that the energy threshold changes between obseravtions.
+To ensure that the `npred` (ie, the predicted number of counts) on the stacked
+dataset is the sum expected by stacking the npred of the individual runs,
+a `~gammapy.irf.EDispersionMap` is used, which contains the
+The mask_safe from each dataset is applied on the respective reconstructed energy axis
+of the energy dispersion matrix, and the masked matrices are combined.
+Values lying outside the safe mask of each dataset are lost.
+
+The following plot shows the individual and stacked attributes for two `SpectrumDataset`
+
+.. plot:: datasets/plot_stack.py
+
+Stacking of multiple datasets is implemented as follows.
+Here, :math:`k` denotes a bin in reconstructed energy,
+:math:`l` a bin in true energy and
+:math:`j` is the dataset number
+
+.. list-table::
+   :widths: 25 25 50
+   :header-rows: 1
+
+   * - Dataset attribute
+     - Behaviour
+     - Implementation
+   * - ``livetime``
+     - Sum of individual livetimes
+     - :math:`\overline{t} = \sum_j t_j`
+   * - ``mask_safe``
+     - True if the pixel is included in the safe data range.
+     - :math:`\overline{\epsilon_k} = \sum_{j} \epsilon_{jk}`
+   * - ``mask_fit``
+     - Dropped
+     -
+   * - ``counts``
+     - Summed in the data range defined by `mask_safe`
+     - :math:`\overline{\mathrm{counts}_k} = \sum_j \mathrm{counts}_{jk} \cdot \epsilon_{jk}`
+   * - ``background``
+     - Summed in the data range defined by `mask_safe`
+     - :math:`\overline{\mathrm{bkg}_k} = \sum_j \mathrm{bkg}_{jk} \cdot \epsilon_{jk}`
+   * - ``exposure``
+     - Summed in the data range defined by `mask_safe`
+     -  :math:`\overline{\mathrm{exposure}_l} = \sum_{j} \mathrm{exposure}_{jl} \cdot \sum_k \epsilon_{jk}`
+   * - ``psf``
+     - Exposure weighted average
+     - :math:`\overline{\mathrm{psf}_l} = \frac{\sum_{j} \mathrm{psf}_{jl} \cdot \mathrm{exposure}_{jl}} {\sum_{j} \mathrm{exposure}_{jl}}`
+   * - ``edisp``
+     - Exposure weighted average, with mask on reconstructed energy
+     - :math:`\overline{\mathrm{edisp}_{kl}} = \frac{\sum_{j}\mathrm{edisp}_{jkl} \cdot \epsilon_{jk} \cdot \mathrm{exposure}_{jl}} {\sum_{j} \mathrm{exposure}_{jl}}`
+   * - ``gti``
+     - Union of individual `gti`
+     -
+It is important to keep in mind that:
+
+- Stacking happens in-place, ie, ``dataset1.stack(dataset2)`` will overwrite ``dataset1``
+- To properly handle masks, it is necessary to stack onto an empty dataset.
+- Stacking only works for maps with equivalent geometry.
+ Two geometries are called equivalent if one is exactly the same as,
+ or can be obtained from a cutout of, the other.
+- A stacked analysis is reasonable only when adding runs taken by the same instrument.
+
+.. _joint:
```

Comment:
I'm not sure this is the right place to explain the difference joint and stacked analysis. I think the datasets documentation is still mostly on technical aspects and API. I think this information should rather go in the overview and a minimal version of it in the glossary...

Reply:
The only issue with putting it in the overview is it comes before people have an actual idea of what stack and joint fit mean. The associated tutorial (I can add one next week) can focus only on the API, but the can/should still explain the idea.

Reply:
I think the main idea of the overview is to introduce new users to these concepts, without explaining them in detail and the link to the sub pages with more information. But I'm fine to keep it mostly as is for now and possibly move content later if we think it leads to clearer structure.